Makefile

```diff
@@ -57,3 +66,87 @@ else
 	@sed -i "32i $$APPDATA_TEXT" flatpak/org.nanuc.Axolotl.appdata.xml
 	@echo Update complete
 endif
+
+build-zkgroup:
+ifdef CARGO_VERSION
+	@echo "Found cargo with version $(CARGO_VERSION)"
+else
+	@echo cargo not found, please install cargo and rust
+	exit 1
+endif
+ifdef GO_VERSION
+	@echo "Found go with version $(GO_VERSION)"
+else
+	@echo go not found, please install go
+	exit 1
+endif
+ifdef GIT_VERSION
+	@echo "Found go with version $(GIT_VERSION)"
+else
+	@echo go not found, please install git
+	exit 1
+endif
+ifeq ($(uname_s), Linux)
+ifeq ($(shell uname --hardware-platform), x86_64)
+	@echo "get zkgroup $(PLATFORM)"
+	go get -d github.com/nanu-c/zkgroup
+	&& git submodule update \
+	&& cd lib/zkgroup \
+	&& cargo build --release --verbose
+	mv libzkgroup.so libzkgroup_linux_amd64.so
+else ifeq ($(shell uname --hardware-platform), aarch64)
```

Comment:
As we are using `$(shell uname --hardware-platform)` two times, is it worth extracting it as a variable alongside of GO_VERSION and the others?